src/components/tools/swot-analysis/steps/SWOTClassifyAlternativeActions/ModalClassifying/ClassifyingCardList/ClassifyingCardList.tsx

```diff
@@ -15,20 +16,18 @@ class ClassifyingCardList extends PureComponent<ClassifyingCardListProps, {}> {
             <>
                 <div className={"actionCards"}>
                     {this.props.actions.map((action, index) => {
-                        return (
-                            <ClassifyingCard
-                                disabled={this.props.disabled}
-                                key={action.name + "-" + action.index + "-" + index}
-                                action={action}
-                                onChangeClick={this.props.onOpenClassificationModalClick}
-                            />
-                        );
+                        return React.cloneElement(this.props.cardElement, {
+                            key: action.name + "-" + action.index + "-" + index,
```

Comment:
I would suggest to remove the optional props of the ClassifyingCard by passing the onChangeClick and disabled props while creating the ClassifyingCardList because the props are there available anyway. Additionally I would suggest to add a default action so this props isn't optional either.

I think it's better when the element is created outside of  this component to pass as much as data from the outside.

Reply:
Another alternative would be to change the props of ClassifyingCard to have the same as ClassifyingMenuCard so that here createElement can be used, instead of clone element.